plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/wizards/NewInterfaceWizardPage.java

```diff
@@ -0,0 +1,173 @@
+		super(referencedComposite);
+	}
+
+	/**
+	 * Sets a given name.
+	 * 
+	 * @param name Name
+	 */
+	public NewInterfaceWizardPage setName(String name) {
+		new LabeledText(this, "Name:").setText(name);
+		return this;
+	}
+
+	/**
+	 * Sets a given package name.
+	 * 
+	 * @param packageName Package name
+	 */
+	public NewInterfaceWizardPage setPackage(String packageName) {
+		new LabeledText(this, "Package:").setText(packageName);
+		return this;
+	}
+
+	/**
+	 * Sets a given source folder.
+	 * 
+	 * @param sourceFolder Source folder
+	 */
+	public NewInterfaceWizardPage setSourceFolder(String sourceFolder) {
+		new LabeledText(this, "Source folder:").setText(sourceFolder);
+		return this;
+	}
+
+	/**
+	 * Returns a package name.
+	 * 
+	 * @return package name
+	 */
+	public String getPackage() {
+		return new LabeledText(this, "Package:").getText();
+	}
+
+	/**
+	 * Returns a interface name.
+	 * 
+	 * @return Interface name
+	 */
+	public String getName() {
+		return new LabeledText(this, "Name:").getText();
+	}
+
+	/**
+	 * Returns a source folder.
+	 * 
+	 * @return Source folder
+	 */
+	public String getSourceFolder() {
+		return new LabeledText(this, "Source folder:").getText();
+	}
+
+	/**
+	 * Check/Uncheck Enclosing type.
+	 *
+	 * @param toggle enclosing type checkbox
+	 */
+	public NewInterfaceWizardPage toggleEnclosingTypeCheckBox(boolean toggle) {
+		new CheckBox("Enclosing type:").toggle(toggle);
+		return this;
+	}
+
+	/**
+	 * Returns Enclosing type.
+	 * 
+	 * @return Enclosing type
+	 */
+	public String getEnclosingType() {
+		return new DefaultText(2).getText();
+	}
+
+	/**
+	 * Sets a given Enclosing type.
+	 * 
+	 * @param enclosingTypeName Enclosing type
+	 */
+	public NewInterfaceWizardPage setEnclosingType(String enclosingTypeName) {
+		new DefaultText(2).setText(enclosingTypeName);
+		return this;
+	}
+
+	/**
+	 * Add extended interface
+	 * 
+	 * @param interfaceName String with name of interface to add
+	 */
+	public void addExtendedInterface(String interfaceName) {
+		new PushButton(3).click();
+		new DefaultText(0).setText(interfaceName);
+
+		switch (new DefaultTable(0).getItems().size()) {
+		case 0:
+			new PushButton(1).click();
+			throw new RedDeerException("No item was found for given interface name '" + interfaceName + "'.");
+		case 1:
+			new PushButton(2).click();
+			break;
+		default:
+			new PushButton(1).click();
+			throw new RedDeerException("More than 1 item was found for given interface name '" + interfaceName + "'.");
+		}
+	}
+
+	/**
+	 * Remove extended interface
+	 * 
+	 * @param interfaceName String with name of interface to remove
+	 */
+	public void removeExtendedInterface(String interfaceName) {
+		DefaultTable table = new DefaultTable(0);
+
+		try {
+			table.getItem(interfaceName).select();
+		} catch (CoreLayerException e) {
+			throw new RedDeerException("No item was found for given interface name '" + interfaceName + "'.");
+		}
+
+		new PushButton(4).click();
```

Comment:
Using indexes on buttons could be tricky, because when the UI changes, like a new button is added, the indexes are different. Using indexes is good practice when you have buttons with same label, but otherwise it is better to use labels(it also helps readability of code).

Reply:
Fixed.